### PR TITLE
Add single Dockerfile for quickstart of new users

### PR DESCRIPTION
**Pull request by bjoern-r**
This pull adds a Dockerfile and a entrypoint scriptto the root of the repository. When these files are present the repository could easily be added to the docker hub as an autobuild project. 
On every change to voctomix git the docker hub will rebuild the public docker container. 
This container can be used to quickly setup a working voctomix "cluster" (core, gui and source-scripts)

### CODE REVIEW

**Comment by bjoern-r** [comment]
# To test the docker version of voctomix use this commands:
## build the docker container locally
- checkout branch: 
```
git checkout quickstart-docker
```
- build docker
```
docker build -t local/voctomix .
```
- rebuild docker after changes
```
docker tag local/voctomix:latest local/voctomix:old; \
docker build -t local/voctomix . && docker rmi local/voctomix:old
```

## Test the docker
the entrypoint script of the container provides some commands to ease the startup of the individual components. get a list by running
```docker run --rm -it --name=voctocore local/voctomix core```

## Run the components
### CORE
```
docker run --rm -it --name=voctocore local/voctomix core
```
### Source example scripts
```
docker run -it --rm --name=cam1 --link=voctocore:corehost local/voctomix ./gstreamer/source-videotestsrc-as-cam1.sh
docker run -it --rm --name=bg --link=voctocore:corehost local/voctomix gstreamer/source-videotestsrc-as-background-loop.sh
```

### GUI
to run the GUI in a docker the docker user needs access to the local X server. This is done by sharing the ```/tmp/.X11-unix``` socket with the container. Depending on your X11 setup you have to allow access to the X-Server session by running: ```xhost +local:$(id -un)```
The example below maps the local voctogui config file ```/tmp/vocto/configgui.ini``` into the container. Please create and change this file to change the voctogui configuration.
```
docker run -it --rm --name=gui --env=gid=$(id -g) --env=uid=$(id -u) --env=DISPLAY=:0 --link=voctocore:corehost \
  -v /tmp/vocto/configgui.ini:/opt/voctomix/voctogui/config.ini -v /tmp/.X11-unix:/tmp/.X11-unix -v /tmp/.docker.xauth:/tmp/.docker.xauth local/voctomix gui
```

**Comment by MaZderMind** [comment]
pretty nice, I'll love to merge this. Two little things:
1. the Dockerfile doesn't have a newline ad the file ending, would id be possible to add this?
2. would you mind linking the README_DOCKER.md from th README.md as another option to get started?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#
+## core:
+# docker run -it --rm -v /some/dir:/video
+#	-p 9999:9999 -p 10000:10000 -p 10001:10001 -p 10002:10002 -p 11000:11000 -p 12000:12000 \
+#	-p 13000:13000 -p 13001:13001 -p 13002:13002 -p 14000:14000 -p 15000:15000 -p 16000:16000 \
+#	-p 17000:17000 -p 17001:17001 -p 17002:17002 -p 18000:18000 --name=voctocore local/voctomix core
+#
+## test sources 
+# docker run -it --rm --name=cam1 --link=voctocore:corehost local/voctomix ./gstreamer/source-videotestsrc-as-cam1.sh
+# docker run -it --rm --name=bg --link=voctocore:corehost local/voctomix gstreamer/source-videotestsrc-as-background-loop.sh#
```
Comment (MaZderMind) [inline Dockerfile]: is there a ./ missing here?

File: README_DOCKER.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## Run the components
+### CORE
+```
+docker run --rm -it --name=voctocore local/voctomix core
+```
+### Source example scripts
+```
+docker run -it --rm --name=cam1 --link=voctocore:corehost local/voctomix ./gstreamer/source-videotestsrc-as-cam1.sh
+docker run -it --rm --name=bg --link=voctocore:corehost local/voctomix gstreamer/source-videotestsrc-as-background-loop.sh
```
Comment (MaZderMind) [inline README_DOCKER.md]: is there a ./ missing here?

File: docker-ep.sh [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+##
+## entrypoint for the docker images
+
+if [ ! -f /.dockerenv ] && [ ! -f /.dockerinit ]; then
+	echo "WARNING: this scrip should be only runed inside docker!!"
```
Comment (MaZderMind) [inline docker-ep.sh]: scrip is missing a t, isn't it?

File: Dockerfile (hunk 1 above)
Comment (bjoern-r) [inline Dockerfile]: it works with and without. the string is executed as a command relative to ```./voctomix/examples/```

**Pull request closed without merging**